ebnf.txt

```diff
@@ -0,0 +1,26 @@
+PROGRAM         ::= ‘project’ (str | STRLIST) nl MODULE? nl (DIR|CLASS|INTERFACE)* nl
+MODULE          ::= ‘modules’ STRLIST nl 				# should have quotes
+DIR             ::= ‘dir’ str nl (ind (DIR|CLASS|INTERFACE) ded)* nl
+CLASS           ::= ABS? ‘class’ str IMP? EXT? (nl COMMENT)? (nl FIELD)* (nl FUNC)* nl
+INTERFACE       ::= ‘interface’ str EXT? (nl COMMENT)? (nl FIELD)* (nl FUNC)* nl
+
+FIELD           ::= ind ‘fields’ mod? VARLIST (nl ind 'generate' gen)* ded
+FUNC            ::= ind ‘function ’ mod ‘async ’? 'static'? str (nl PARAM)? (nl ind 'returns' type)? (nl ind COMMENT)? ded
+
+PARAM           ::= ind ‘params ’ VARLIST ded
+COMMENT         ::= ind ‘comments ’ STRLIST ded
+
+ABS             ::= ‘abstract’
+IMP             ::= ‘implements’ str (‘,’ str)*
+EXT             ::= ‘extends’ str
+VARLIST         ::= ‘[’ (type str (‘,’ type str))? ‘]’  # I think I did this
```

Comment:
I think comment left from copy/paste here, which can be removed.

Reply:
Fixed